internal/cephes/igam.go

```diff
@@ -0,0 +1,378 @@
+	[igamDimN]float64{7.389033153567425e+1, -1.5680141270402273e+2, 1.322177542759164e+2, 1.3692876877324546e-2, -1.2366496885920151e+2, 1.4620689391062729e+2, -8.0365587724865346e+1, -1.1259851148881298e-4, 4.0770132196179938e+1, -3.8210340013273034e+1, 1.719522294277362e+1, 9.3519707955168356e-7, -6.2716159907747034, 5.1168999071852637, -2.0319658112299095, -4.9507215582761543e-9, 5.9626397294332597e-1, -4.4220765337238094e-1, 1.6079998700166273e-1, -2.4733786203223402e-8, -4.0307574759979762e-2, 2.7849050747097869e-2, -9.4751858992054221e-3, 6.419922235909132e-6, 2.1250180774699461e-3},
+	[igamDimN]float64{2.1216837098382522e+2, 1.3107863022633868e+1, -4.9698285932871748e+2, 7.3121595266969204e+2, -4.8213821720890847e+2, -2.8817248692894889e-2, 3.2616720302947102e+2, -3.4389340280087117e+2, 1.7195193870816232e+2, 1.4038077378096158e-4, -7.52594195897599e+1, 6.651969984520934e+1, -2.8447519748152462e+1, -7.613702615875391e-7, 9.5402237105304373, -7.5175301113311376, 2.8943997568871961, -4.6612194999538201e-7, -8.0615149598794088e-1, 5.8483006570631029e-1, -2.0845408972964956e-1, 1.4765818959305817e-4, 5.1000433863753019e-2, -3.3066252141883665e-2, 1.5109265210467774e-2},
+	[igamDimN]float64{-9.8959643098322368e+2, 2.1925555360905233e+3, -1.9283586782723356e+3, -1.5925738122215253e-1, 1.9569985945919857e+3, -2.4072514765081556e+3, 1.3756149959336496e+3, 1.2920735237496668e-3, -7.525941715948055e+2, 7.3171668742208716e+2, -3.4137023466220065e+2, -9.9857390260608043e-6, 1.3356313181291573e+2, -1.1276295161252794e+2, 4.6310396098204458e+1, -7.9237387133614756e-6, -1.4510726927018646e+1, 1.1111771248100563e+1, -4.1690817945270892, 3.1008219800117808e-3, 1.1220095449981468, -7.6052379926149916e-1, 3.6262236505085254e-1, 2.216867741940747e-1, 4.8683443692930507e-1},
+}
+
+// Igam computes the incomplete Gamma integral
+func Igam(a, x float64) float64 {
+	// Check zero integration limit first
+	if x == 0 {
+		return 0
+	}
+
+	if x < 0 || a <= 0 {
+		panic(badParamOutOfBounds)
+	}
+
+	// Asymptotic regime where a ~ x; see [2].
+	absxmaA := math.Abs(x-a) / a
+	if (a > small && a < large && absxmaA < smallRatio) ||
+		(a > large && absxmaA < largeRatio/math.Sqrt(a)) {
+		return asymptoticSeries(a, x, igam)
+	}
+
+	if x > 1 && x > a {
+		return 1 - IgamC(a, x)
+	}
+
+	return igamSeries(a, x)
+}
+
+// IgamC computes the complemented incomplete Gamma integral
+func IgamC(a, x float64) float64 {
+	if x < 0 || a <= 0 {
+		panic(badParamOutOfBounds)
+	} else if x == 0 {
+		return 1
+	} else if x == math.MaxFloat64 || x == -math.SmallestNonzeroFloat64 {
```

Comment:
Why are you using these math constants rather than an inf check? It's not clear these particular values need special treatment.

If it's for immediate consistency, maybe add a TODO to change later?